client/src/components/Grid/configs/histories.ts

```diff
@@ -71,6 +71,8 @@ const batch: BatchOperationArray = [
                 try {
                     const historyIds = data.map((x) => String(x.id));
                     await deleteHistories({ ids: historyIds });
+                    const historyStore = useHistoryStore();
+                    await historyStore.handleTotalCountChange(data.length, true);
```

Comment:
All these `handleTotalCountChange` in this file make me think these calls (delete/undelete/etc.) should be made through the store, not directly calling the API. So the internal `handleTotalCountChange` can still be internal and handled accordingly after each operation.

Reply:
I agree, that was my first thought as well. Let me instead do all this through the store in the grid config as well then.

Reply:
Moved these calls to the history store itself, and it works nicely and the selector modal is synced with the changes made in the grid as well

Reply:
Thank you!